compiler/z/codegen/BinaryEvaluator.cpp

```diff
@@ -1045,7 +1052,8 @@ iDivRemGenericEvaluator(TR::Node * node, TR::CodeGenerator * cg, bool isDivision
    //    R1+1 -> Quotient
    //
    // Note:  D and DR require that R1=0 before executing the division
-   bool needCheck = (!node->isSimpleDivCheck()            &&
+   // TODO(#3686): Remove this Java-ism from OMR and push it into OpenJ9
+   bool needCheck = (!node->getOpCode().isUnsigned() && !node->isSimpleDivCheck()            &&
```

Comment:
pretty minor point: wouldn't it make more sense in this change and the previous one to add the `!isUnsigned` check next to the `!isNonNegative` check rather than adding it "next" to whatever the first condition happens to be?

Reply:
Fixed in 36995ad